src/legacy-appcache-behavior-import.js

```diff
@@ -0,0 +1,422 @@
+/* eslint-env worker, serviceworker */
+
+(function(global) {
+  'use strict';
+  // Code in the ServiceWorkerGlobalScope can safely assume that a greater
+  // set of ES2015 features are available, without having to transpile.
+
+  const log = console.debug.bind(console);
+
+  const constants = require('./lib/constants.js');
+  const idb = require('./lib/idb-helpers.js');
+
+  /**
+   * Determines what the most likely URL is associated with the client page from
+   * which the event's request originates. This is used to determine which
+   * AppCache manifest's rules should be applied.
+   *
+   * @param {FetchEvent} event
+   * @returns {String} The client URL
+   */
+  function getClientUrlForEvent(event) {
+    // If our service worker implementation supports client identifiers, try
+    // to get the client URL using that.
+    if (global.clients && global.clients.get && event.clientId) {
+      return global.clients.get(event.clientId).then(client => client.url);
+    }
+
+    // Otherwise, try to get the client URL using the Referer header.
+    // And if that's not set, assume that it's a navigation request and the
+    // effective client URL should be the request URL.
+    return Promise.resolve(event.request.referrer || event.request.url);
+  }
+
+  /**
+   * Finds the longest matching prefix, given an array of possible matches.
+   *
+   * @param {Array.<String>} urlPrefixes
+   * @param {String} fullUrl
+   * @returns {String} The longest matching prefix, or '' if none match
+   */
+  function longestMatchingPrefix(urlPrefixes, fullUrl) {
+    return urlPrefixes
+      .filter(urlPrefix => fullUrl.startsWith(urlPrefix))
+      .reduce((longestSoFar, current) => {
+        return longestSoFar.length >= current.length ? longestSoFar : current;
+      }, '');
+  }
+
+  /**
+   * Performs a fetch(), using a cached response as a fallback if that fails.
+   *
+   * @param {Request} request
+   * @param {String} fallbackUrl
+   * @param {String} cacheName
+   * @returns {Promise.<Response>}
+   */
+  function fetchWithFallback(request, fallbackUrl, cacheName) {
+    log('Trying fetch for', request.url);
+    return fetch(request).then(response => {
+      // Succesful but error-like responses are treated as failures.
+      // Ditto for redirects to other origins.
+      if (!response.ok || (new URL(response.url).origin !== location.origin)) {
+        throw Error('Fallback request failure.');
+      }
+      return response;
+    }).catch(() => {
+      log('fetch() failed. Falling back to cache of', fallbackUrl);
+      return caches.open(cacheName).then(
+        cache => cache.match(fallbackUrl));
```

Comment:
I know you are not dealing with errors for version 0.0.1 but notice [`match()` does not reject if there is no match](https://developer.mozilla.org/en-US/docs/Web/API/Cache/match). If you want to convert a miss into a rejection for further handling, make this as follows:
```js
cache => cache.match(fallbackUrl).then(match => {
  if (!match) { throw Error('Missing fallback'); }
  return match;
})
```